### PR TITLE
feature-robo-control

### PR DESCRIPTION
**Pull request by LachlanMurphy**
Moved around some items in the main file and created the control_manager.hpp/cpp. We are now good to start working on different control methods, though we do also need to think about how to integrate the yaml into the control manager file.

### CODE REVIEW

File: src/controls/control_manager.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+#include "../comms/rm_CAN.hpp"
+
+/// @brief enums for each control type
+enum control_types {
+    PID = 0,
+    // ... the rest of the control types
+};
+
+class control_manager {
```
Comment (Pandabear1125) [inline src/controls/control_manager.hpp:12]: Rename this to ControlManager to follow style guide

Comment (Pandabear1125) [reply src/controls/control_manager.hpp:12]: Also, doxygen is throwing undocumentation warnings at you for control_manager btw

File: src/controls/control_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+control_manager::control_manager(rm_CAN* _can) {
+    can = _can;
+}
+
+void control_manager::step_motors() {
+    // update the data that we have
+
+    // runs the set control method on each motor
+    for (int i = 0; i < NUM_MOTORS; i++) {
+
```
Comment (nitroxplunge) [inline src/controls/control_manager.cpp:12]: Remove all this whitespace

File: src/controls/control_manager.hpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
+#ifndef CONTROL_MANAGER_HPP
+#define CONTROL_MANAGER_HPP
+
+#include "../comms/rm_CAN.hpp"
+
+/// @brief enums for each control type
+enum control_types {
```
Comment (nitroxplunge) [inline src/controls/control_manager.hpp:7]: This is awesome

**Review by nitroxplunge — APPROVED** [review]

**Comment by Pandabear1125** [comment]
Marking this as closed. The main robot control code was merged with feature-controllers. This is now obsolete.

**Pull request closed without merging**